### PR TITLE
Unit tests

### PR DESCRIPTION
**Pull request by ticoniq**
## Unit tests

This pull request aims to enhance the Object-Oriented Programming (OOP) school library project by adding comprehensive unit tests using RSpec. These tests have been designed to verify the correctness and reliability of our library classes and methods, ensuring that our codebase remains robust and maintainable.

### Changes Made

1. **Unit Test Coverage**: I have created a series of RSpec unit tests to cover the core functionality of our library classes, including Student, Teacher, Book, Rental, and any other relevant classes. These tests thoroughly assess each class's behavior and edge cases.

2. **RSpec Configuration**: I have set up the necessary RSpec configuration files, such as `spec` to ensure a consistent and organized testing environment.

3. **Unit Tests**: The unit tests focus on individual components, checking that each class and method behaves as expected and handles various scenarios correctly.

### CODE REVIEW

File: spec/person_spec.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
+require_relative 'spec_helper'
+
+describe Person do
+  before(:each) do
+    @person = Person.new(25, 'John Doe', parent_permission: true)
+  end
+
```
Comment (BregornOriginal) [inline spec/person_spec.rb:8]: - [ ] Kindly add test for your method `add_rental` in your person_spec.rb file

**Review by BregornOriginal — CHANGES_REQUESTED** [review]
Hi @Team 

Good job so far!

There are some issues that you still need to work on to go to the next project but you are almost there!

giphy (4)

## Highlights

✅ Linters are passing
✅ Good commit history
✅ README looks professional 🥇 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Reem-lab — APPROVED** [review]
# Hi Team,  👋🏻👋🏻

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights

- [ ]  Followed Gitflow 💯
- [ ]  Descriptive PR⭐
- [ ]  Kudos for working on the tests🤩
- [ ]  No Linters error ✅
- [ ] All tests passed 🎉



## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by migo1** [comment]
@reem-lab thanks for the reviews
